multidict/_multidict_py.py

```diff
@@ -628,9 +631,13 @@ def _extend(
     ) -> None:
         if arg:
             if isinstance(arg, (MultiDict, MultiDictProxy)):
-                items = arg._impl._items
+                if self._ci != arg._ci:
```

Comment:
it's customary to use identity checks for booleans
```suggestion
                if self._ci is not arg._ci:
```